### DISCUSSION TITLE
Allow throttling of pylance errors while typing

### DISCUSSION THREAD

**Discussion by el1s7** [microsoft/pylance-release#6787] [Enhancement]
I don't know if something changed this past few months in this extension or VS Code in general (or if it's just me xD), but the error hints seem to somehow be getting more annoying.

Like, I still haven't finished typing a line of code but I will instantly get a red line hovering on anything while I'm still typing, not only annoying but also distracting. Example (notice the instant red underline):

![Image](https://github.com/user-attachments/assets/5853556a-1a97-48ae-b0f7-5bf3fabbbc91)

### It would be really good if there were some configuration options  such as:

-  `diagnosticReportDelay`: to set a custom delay/debounce timeout for diagnostic error reporting (i.e. how many ms after typing)
-  `dianosticReportOnlyOnSave`: to only show the error reporting when the file is saved.

It seems that there was an [option](https://github.com/microsoft/python-language-server/issues/1944) before for setting the delay in the old [microsoft/pylance-release](https://github.com/microsoft/python-language-server/issues/1944) (namely `diagnosticPublishDelay`), but there's no option anywhere now.

For comparison, if we look at [this actual repo showcase GIF](https://github.com/microsoft/pylance-release/blob/main/images/all-features.gif) in readme from 3 years ago, you can notice that there is a bit longer delay before the warning underline report as you type. 

Also, I just noticed that if I disable Pylance, and VS Code fallbacks to Jade, then there's a bit longer delay as well.

**Comment by el1s7** [comment]
@rchiodo @KacieKK can we have a configuration option for setting the delay/debounce for error reporting? 

It was previously a supported configuration at the old [python-language-server](https://github.com/microsoft/python-language-server/issues/1944).

**Comment by rchiodo** [comment]
I _think_ this should be relatively easy to support. Analysis already has a throttle here:
https://github.com/microsoft/pyright/blob/1f5a2bcdd8967804dba55762b5dbbe735867fece/packages/pyright-internal/src/analyzer/service.ts#L71, but I'm not sure if that applies to parse errors. The errors you're seeing are parse/syntax errors for invalid code. 

That shouldn't be too hard to throttle either. Transferring issue to enhancements for upvotes.
